Source/WTF/wtf/TypeTraits.h

```diff
@@ -92,25 +92,40 @@ struct RemoveSmartPointerHelper<T, Ref<Pointee>> {
 template<typename T>
 struct RemoveSmartPointer : detail::RemoveSmartPointerHelper<T, std::remove_cv_t<T>> { };
 
-// HasRefCountMethods implementation
+// HasRefPtrMethods implementation
 namespace detail {
 
 template<typename>
 struct SFINAE1True : std::true_type { };
 
 template<class T>
-static auto HasRefCountMethodsTest(int) -> SFINAE1True<decltype(std::declval<T>().ref(), std::declval<T>().deref())>;
+static auto HasRefPtrMethodsTest(int) -> SFINAE1True<decltype(&T::ref, &T::deref)>;
```

Comment:
Do we really need int argument here?

Reply:
The int vs long argument overload is one of the -- admittedly very weird -- customary ways to do C++ SFINAE programming. 'int' is preferred over 'long' in overload resolution, so the compiler will select the 'int' version if it is not an error, and otherwise select the 'long' version. So, in a very weird way, 'int' means "use this if possible" and 'long' means "otherwise default to this if necessary".

Reply:
oh, I see. That's pretty counterintuitive. Maybe we should add a comment there explaining as such. Or maybe in the commit message?